doc/examples/example_ecies.md

```diff
@@ -0,0 +1,378 @@
+the algorithms used for symmetric encryption, key derivation and message authentication can be configured
+in out implementation. For the sake of easy usage, we have decided to use the following defaults:
+ - Symmetric Encryption: AES-CBC with PKCS7 padding, IV=0x0, keysize 256 bit
+ - Key Derivation Function: X9.63 without salt
+ - Message Authentication Code: HMAC-SHA512 without salt
+
+The following examples show how to en- and decrypt using ECIES with these default settings. Afterwards, the
+different options to change the default values will be shown.
+
+## Encryption
+
+```cpp
+// 1. Read public key from PEM
+mococrw::AsymmetricPublicKey pubKey = mococrw::AsymmetricPublicKey::readPublicKeyFromPEM(secp384PubKeyPEM);
+
+std::vector<uint8_t> testString = {'H', 'e', 'l', 'l', 'o', ' ', 'W', 'o', 'r', 'l', 'd', '!'};
+
+mococrw::ECIESCtxBuilder encBuilder;
+std::unique_ptr<mococrw::ECIESEncryptionCtx> encCtx = encBuilder.buildEncryptionCtx(pubKey);
+encCtx->update(testString);
+
+// 2. Finish encryption and extract parameters for decryption
+std::vector<uint8_t> ciphertext = encCtx->finish();
+std::vector<uint8_t> mac = encCtx->getMAC();
+std::vector<uint8_t> point = encCtx->getEphemeralKey().toECPoint(...); // See section "Ephemeral Key Encodig Format" for details
+```
+
+## Decryption
+
+```cpp
+// Ciphertext, mac and the point need to be available
+std::vector<uint8_t> ciphertext = { ... };
+std::vector<uint8_t> mac = { ... };
+std::vector<uint8_t> point = { ... };
+
+// 1. Read private key from PEM
+mococrw::AsymmetricKeypair privKey = mococrw::AsymmetricKeypair::readPrivateKeyFromPEM(secp384KeyPEM, "password");
+
+// 2. Get type of curve from private key
+std::shared_ptr<mococrw::ECCSpec> spec;
+{
+    // Convert returned unique_ptr into shared_ptr
+    std::shared_ptr<mococrw::AsymmetricKey::Spec> spec_tmp = privKey.getKeySpec();
+
+    // Cast shared_ptr<mococrw::AsymmetricKey::Spec> into std::shared_ptr<mococrw::ECCSpec>
+    spec = std::dynamic_pointer_cast<mococrw::ECCSpec>(spec_tmp);
+}
+
+if (!spec) {
+    // Error handling
+    ...
+}
+
+// 3. Re-create ephemeral key
+mococrw::AsymmetricPublicKey ephKey = mococrw::AsymmetricPublicKey::fromECPoint(spec, point);
+
+mococrw::ECIESCtxBuilder decBuilder;
+
+// 4. Set parameters for decryption
+std::unique_ptr<mococrw::ECIESDecryptionCtx> decCtx = decBuilder.buildDecryptionCtx(privKey, ephKey);
+decCtx->update(ciphertext);
+decCtx->setMAC(mac);
+
+// 5. Decrypt
+try {
+    std::vector<uint8_t> result = decCtx->finish();
+}
+catch (const mococrw::MoCOCrWException &e)  {
+    std::cerr << "MAC verification failed" << std::endl;
+    ...
+}
+```
+
+## Serialization of Artifacts
+
+The encrption of the plaintext creates several artiacts (ciphertext, mac and ephemeral key) that need to be transfered to the client in order to
+decrypt the ciphertext. MoCOCrW offer no serilaization functionality because we don't want to make any assumptions of the format. That is,
```

Comment:
```suggestion
decrypt the ciphertext. MoCOCrW offers no serialization functionality because we don't want to make any assumptions of the format. That is,
```